webapp/src/utils/Promises.ts

```diff
@@ -0,0 +1,11 @@
+export class Promises<T> {
```

Comment:
for this class I recommend to:
1. rename it to something not to Promise like AsyncTasks, for example var promises was hard to know if it represent Promise or Promises
2. use static factory, give more control in future and easier to use than write keyword "new" everytime
3. rename function callbackfn -> callbackFn

suggestion:

```
export class AsyncTasks<T> {
  private constructor(private promises: Array<Promise<T>>) {}

  static of<T>(promises: Array<Promise<T>>) {
    return new AsyncTasks(promises);
  }

  map<U>(callbackFn: (value: T) => U | Promise<U>): AsyncTasks<U> {
    return new AsyncTasks(this.promises.map((p) => p.then(callbackFn)));
  }

  all() {
    return Promise.all(this.promises);
  }
}


```

Reply:
I have wrapped the constructor in a factory to avoid precedence ambiguity like `new X().Y()` and I have renamed the parameter to align with MDN.

We discussed over Slack what to do with the confusing class name. `Tasks` is not an excellent name as Promises more represents the promised result than the task itself. `Promises` is pretty good as it describes the contents of objects of this class. We resolved the confusion by renaming the field instead.